### PR TITLE
test(review): cover review edge cases

### PR DESCRIPTION
**Pull request by hirotomoyamada**
## AI used

- [x] I checked the generated content before submitting.

## Description

Adds focused review edge-case coverage and removes redundant validation guards.

## Current behavior (updates)

Several reachable review safety, merge-queue, and Git diff parsing paths were not tested.

## New behavior

The paths are covered by unit tests, including merge-queue conflict after queue departure.

## Is this a breaking change (Yes/No):

No

## Additional Information

- `pnpm quality` passes.

### CODE REVIEW

**Review by caspar-magi — APPROVED** [review]

**Review by balthasar-magi — APPROVED** [review]

File: src/tools/review/reviewer.ts [hunk 1]
```diff
@@ -484,11 +484,6 @@ async function collectAcceptedFindings(
 
   if (!findings.length) return accepted
```
Comment (melchior-magi) [inline src/tools/review/reviewer.ts:485]: Missing changeset file. The removal of the redundant `No reviewers configured.` / `Reviewers not found.` guards inside `collectAcceptedFindings` modifies a non-test source file in `src/**`, so per the changeset rules this PR needs a `.changeset/*.md` entry.

**Review by melchior-magi — CHANGES_REQUESTED** [review]
The PR adds review edge-case test coverage (branch-mismatch safety, non-JSON quoted diff path, merge-queue conflict after queue departure, retry exhaustion, and the `requires %s` guards inside `validateFindings`) and removes a pair of redundant validation guards inside `collectAcceptedFindings`. The same guards remain enforced by the caller `validateFindings`, and the new `requires %s` test at lines 1791-1803 confirms that behavior. The PR modifies a non-test source file but does not include the changeset that `.agents/rules/changesets.md` requires for `src/**` changes; prior similar refactors (commits `05c255f` and `6535875`) include one.

File: src/tools/review/reviewer.ts (hunk 1 above)
Comment (hirotomoyamada) [reply src/tools/review/reviewer.ts:485]: Fixed.

**Review by balthasar-magi — APPROVED** [review]

**Review by caspar-magi — APPROVED** [review]

**Review by melchior-magi — APPROVED** [review]
